src/dsp_tools/utils/id_to_iri.py

```diff
@@ -73,16 +76,35 @@ def _replace_ids_by_iris(
         modified XML tree
     """
     success = True
-    resource_elements = tree.xpath("/knora/resource/resptr-prop/resptr")
-    for resptr_prop in resource_elements:
-        value_before = resptr_prop.text
-        value_after = mapping.get(resptr_prop.text)
-        if value_after:
-            resptr_prop.text = value_after
-        else:
-            logger.warning(f"Could not find internal ID '{value_before}' in mapping file. Skipping...")
-            print(f"WARNING: Could not find internal ID '{value_before}' in mapping file. Skipping...")
-            success = False
+    used_mapping_entries = set()
+
+    resptr_elems = tree.xpath("/knora/resource/resptr-prop/resptr")
+    resptr_elems_replaced = 0
+    for resptr_elem in resptr_elems:
+        value_before = resptr_elem.text
+        if value_after := mapping.get(value_before):
+            resptr_elem.text = value_after
+            resptr_elems_replaced += 1
+            used_mapping_entries.add(value_before)
+
+    salsah_links = [
+        x for x in tree.xpath("/knora/resource/text-prop/text//a") if x.attrib.get("class") == "salsah-link"
+    ]
+    salsah_links_replaced = 0
```

Comment:
I would make salsah and resptr_elems two separate functions.

Reply:
better now?